src/teleport.h

```diff
@@ -0,0 +1,17 @@
+#pragma once
+#ifndef TELEPORT_H
+#define TELEPORT_H
+
+#include "creature.h"
```

Comment:
Not needed. A forward declaration should will do just fine.
```suggestion
class Creature;
```